pyproject.toml

```diff
@@ -0,0 +1,10 @@
+[build-system]
```

Comment:
> I have made the reason for my objection clear before: Both this file (and the introduction of the file `meson.build` in SAGE_ROOT, as discussed in [#36489 (comment)](https://github.com/sagemath/sage/pull/36489#discussion_r1375152978)) mislead users into assuming that they can build Sage using `pip install .` and/or `meson ...` instead of using `./configure && make`. This is not true because the latter are the build system of the Sage **distribution**, whereas you have developed a build system for the (monolithic) Sage **library** -- with is the Python project already declared in `src/pyproject.toml` named "sagemath-standard".

I've developed an alternative for make for the sage project. In this meson alternative, there is not such a distinction between sage-the-distribution and sage-library. There is one project, called sage (or sagemath if you prefer). You can build it using `pip install .` (plus some args) or using `meson compile` (plus some args) from the root. The only difference to the current `make build` command is that the developer is responsible for setting up the surrounding build environment (i.e. if dependencies are not present, the build fails). This is very much similar to scipy/numpy/name another meson project.
 
> The correct solution is to edit `src/pyproject.toml` so that the Sage library uses the mesonpy build system instead of setuptools (delete setup.py), and to use `src/meson.build` as the toplevel meson file. This affects the default (monolithic editable) build. Users can use `./configure --disable-editable` or `./configure --disable-editable --enable-wheels` to switch back to a setuptools-based build system.

I don't think the meson-based system can directly function as a new default. It's unrealistic that it will directly as stable as the existing setuptools-based one. So it should be opt-in instead of opt-out. So the root directory is the only option I see to place the new pyprojecttoml file.

Reply:
> I've developed an alternative for make for the sage project. In this meson alternative, there is not such a distinction between sage-the-distribution and sage-library.

Try to explain how what you are doing is different from build the Sage library.

Reply:
> The only difference to the current `make build` command is that the developer is responsible for setting up the surrounding build environmen

Does this not mean that it is doing nothing that the Sage distribution does in addition to the Sage library?

Reply:
> [...] So the root directory is the only option I see to place the new pyprojecttoml file.

Happy to help: 
- Option 1: An environment variable could switch `src/pyproject.toml` from using setuptools to using mesonpy. Trivial to do by declaring a custom PEP-517/660 build api (about 10 lines of code); reference: #29845 
- Option 2: A directory `pkgs/sagemath-standard_meson` with pyproject.toml.m4

Reply:
> > I've developed an alternative for make for the sage project. In this meson alternative, there is not such a distinction between sage-the-distribution and sage-library.
> 
> Try to explain how what you are doing is different from building the Sage library.

Yes of course its building the sage library....that's also the end-goal of sage-the-distribution.

Sage-the-library doesn't have any means to detect and link libraries that are needed for its compilation (except a few runtime fallbacks). Finding and setting these dependencies up is currently the job of sage-the-distribution. This part of sage-the-distribution is covered by meson.

Reply:
> Sage-the-library doesn't have any means to detect and link libraries that are needed for its compilation

That's false. It is using pkg-config for this purpose.